secret_programmatic_api_keys.go

```diff
@@ -278,7 +278,15 @@ func (b *Backend) pathProgrammaticAPIKeyRollback(ctx context.Context, req *logic
 
 		// we need the orgID to delete the Key
 		foundKey := mongodbatlas.APIKey{}
-		keys, _, err := client.ProjectAPIKeys.List(ctx, entry.ProjectID, nil)
+
+		// define the list options to get all the keys
+		// currently pull 500 at a time, which is the max according to the docs
+		// https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Programmatic-API-Keys/operation/listProjectApiKeys
+		opts := &mongodbatlas.ListOptions{
+			ItemsPerPage: 500,
```

Comment:
Is it possible to have more than 500 keys? If yes, should we check the `totalCount` in the response and request the next page(s)?

Reply:
Actually, it looks like `totalCount` only tells us how many items are contained in the _current_ response. So we may have to request the next page to check if there are any more items.

Reply:
According to the [mongo docs](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Programmatic-API-Keys), the max we can request is 500.

Reply:
Got it, thanks!